tiledb/sm/query/dense_reader.h

```diff
@@ -111,6 +114,9 @@ class DenseReader : public ReaderBase, public IQueryStrategy {
   /*         PRIVATE ATTRIBUTES        */
   /* ********************************* */
 
+  /** UID of the logger instance */
+  inline static uint64_t logger_id_ = 0;
```

Comment:
The static variables in all the classes have a problem with concurrent access/incrementing right? These need a different way to handle the IDs in order to have concurrent creation?

Reply:
right, fixed it to use atomic static variables instead